mmv1/third_party/terraform/acctest/resource_inventory_reader.go

```diff
@@ -0,0 +1,249 @@
+				return nil // Continue to next file
+			}
+
+			// Parse YAML
+			var metadata ResourceYAMLMetadata
+			if err := yaml.Unmarshal(content, &metadata); err != nil {
+				return nil // Continue to next file
+			}
+
+			// Skip if resource is empty
+			if metadata.Resource == "" {
+				return nil
+			}
+
+			// Store API service name in cache
+			if metadata.ApiServiceName != "" {
+				apiServiceNameCache[metadata.Resource] = metadata.ApiServiceName
+				apiNameCount++
+			}
+
+			// Extract and store service package in cache
+			pathParts := strings.Split(path, string(os.PathSeparator))
+			servicesIndex := -1
+			for i, part := range pathParts {
+				if part == "services" {
+					servicesIndex = i
+					break
+				}
+			}
+
+			if servicesIndex >= 0 && len(pathParts) > servicesIndex+1 {
+				servicePackage := pathParts[servicesIndex+1] // The part after "services"
+				servicePackageCache[metadata.Resource] = servicePackage
+				servicePkgCount++
+			}
+		}
+		return nil
+	})
+
+	if err != nil {
+		return fmt.Errorf("error walking directory: %v", err)
+	}
+
+	// Mark cache as populated
+	cachePopulated = true
+
+	return nil
+}
+
+// GetCacheStats returns statistics about the metadata cache
+func GetCacheStats() (apiServiceNameCount int, servicePackageCount int) {
+	cacheMutex.RLock()
+	defer cacheMutex.RUnlock()
+
+	return len(apiServiceNameCache), len(servicePackageCache)
+}
+
+// GetAPIServiceNameForResource finds the api_service_name for a given resource name
+// If projectRoot is empty, it will attempt to find the project root automatically
+func GetAPIServiceNameForResource(resourceName string) string {
+	// Make sure cache is populated
+	if !cachePopulated {
+		if err := PopulateMetadataCache(); err != nil {
+			return "failed_to_populate_metadata_cache"
+		}
+	}
+
+	// Check cache
+	cacheMutex.RLock()
+	apiServiceName, found := apiServiceNameCache[resourceName]
+	cacheMutex.RUnlock()
+
+	if !found {
+		return "unknown"
+	}
+
+	return apiServiceName
+}
+
+// GetServicePackageForResourceType finds the service package for a given resource type
+// If projectRoot is empty, it will attempt to find the project root automatically
+func GetServicePackageForResourceType(resourceType string) string {
+	// Make sure cache is populated
+	if !cachePopulated {
+		if err := PopulateMetadataCache(); err != nil {
+			return "failed_to_populate_metadata_cache"
+		}
+	}
+
+	// Check cache
+	cacheMutex.RLock()
+	servicePackage, found := servicePackageCache[resourceType]
+	cacheMutex.RUnlock()
+
+	if !found {
+		return "unknown"
+	}
+
+	return servicePackage
+}
+
+// getServicesDir returns the path to the services directory
+// If projectRoot is provided, it will use that as the base
+// Otherwise, it will attempt to find the project root, and if that fails,
+// it will use environment variables or fall back to the current directory
+func getServicesDir(projectRoot string) (string, error) {
+	// If project root is provided, use it
+	if projectRoot != "" {
```

Comment:
When calling getServicesDir, projectRoot is "".

Is the if statement necessary?